src/benchmarks/micro/libraries/System.IO/UnmanagedMemoryStreamTests.cs

```diff
@@ -0,0 +1,53 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.IO;
+using System.Runtime.InteropServices;
+using BenchmarkDotNet.Attributes;
+using MicroBenchmarks;
+
+namespace System.IO.Tests
+{
+    [BenchmarkCategory(Categories.Libraries)]
+    public unsafe class UnmanagedMemoryStreamTests
+    {
+        private const int Length = 10000;
+        private byte* buffer;
+
+        [GlobalSetup]
+        public void Setup()
+        {
+            buffer = (byte*)Marshal.AllocCoTaskMem(Length);
+        }
+
+        [GlobalCleanup]
+        public void Cleanup()
+        {
+            Marshal.FreeCoTaskMem((IntPtr)buffer);
+        }
+
+        [Benchmark]
+        public void ReadAsBytes()
+        {
+            using (var ums = new UnmanagedMemoryStream(buffer, Length))
```

Comment:
I am not sure whether this should better be in IterationSetup. The docs for [IterationSetup](https://github.com/dotnet/BenchmarkDotNet/blob/e170684208103ca5ba4212ad8dc7c2aad5cf02d4/src/BenchmarkDotNet.Annotations/Attributes/IterationSetupAttribute.cs#L7) say that it should not be used for microbenchmarks.

Reply:
As long as the benchmark takes roughly more than 100ms the IterationSetup is fine.

Reply:
This microbenchmark takes a lot less than 100ms and I do not think it makes sense to artificially inflate it to take more than 100ms.